includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -155,6 +155,13 @@ public function capture_terminal_payment( WP_REST_Request $request ) {
 				);
 			}
 
+			// Do not process already paid orders to prevent double-charging.
+			$stored_intent_id     = $order->get_meta( WC_Payments_Order_Service::INTENT_ID_META_KEY );
+			$stored_intent_status = $order->get_meta( WC_Payments_Order_Service::INTENTION_STATUS_META_KEY );
+			if ( ! empty( $stored_intent_id ) && ( Intent_Status::SUCCEEDED === $stored_intent_status || $stored_intent_id !== $intent_id ) ) {
```

Comment:
I'd not make status check dependent on stored intent id and re-worked condition a bit:

`Intent_Status::SUCCEEDED === $stored_intent_status || ($stored_intent_id && $stored_intent_id !== $intent_id)`

Reply:
Can other statuses be stored in the meta? Would it mak more reliable perhaps to check for more [statuses](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-status) like processing and cancelled?

Reply:
It's a valid concern. Attempted to address in ef3d8a9 and fb0ca7b.

Reply:
Thanks!